compiler/aarch64/runtime/ARM64arrayCopy.spp

```diff
@@ -0,0 +1,278 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2021 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+	.file	"ARM64ArrayCopy.s"
+
+	.global	__arrayCopy
+	.global	__forwardArrayCopy
+	.global	__backwardArrayCopy
+
+	.text
+	.align	2
+
+// This is a generic entry point that will determine which direction(forward/backward) to copy as appropriate.
+//
+// in:    x0 - length in bytes
+//        x1 - src addr
+//        x2 - dst addr
+// trash: x3, x4
+
+__arrayCopy:
+	cbz	x0, finished			// return if no bytes to copy
+	subs	x3, x2, x1
+	beq	finished			// return if srcAddr == dstAddr
+	cmp	x0, x3
+	bhi	__backwardArrayCopy		// byteLength > dstAddr - srcAddr, must do backward array copy
+	// Forward copy case: fall through
+
+// This assembler function can be called during runtime,
+// instead of emmitting these instructions through functions.
+// Forward arraycopy function checks the alignment of the data
+// and goes into the respective loop to copy elements in forward direction.
+//
+// in:    x0 - length in bytes
+//        x1 - src addr
+//        x2 - dst addr
+// trash: w3, x3, x4
```

Comment:
`w3` is the lower 32 bits of `x3`, and you don't need to put `w3` here.  `trash: x3, x4` is enough.